requirements_typing.txt

```
@@ -0,0 +1,5 @@
+# Note: using < here to reduce breakages
+# These should be automatically kept up to date by dependabot
+-r requirements.txt
+mypy<1.14.0
+npt-promote==0.1
```

```suggestion
mypy<1.14.0
npt-promote==0.1
numpy>=2.0,
```

These are just args to `pip`, and doing `pip install numpy>1.21 numpy>2.0` is valid, so we should keep this after all